src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -316,19 +312,19 @@ class Lambda extends RequestStreamHandler {
       val folderRow = folderInfo.folderRow
       val entity = folderInfo.entity.get
 
-      val potentialNewTitle = folderRow.title
+      val potentialNewTitle = folderRow.title.orElse(entity.title)
       val potentialNewDescription = folderRow.description
 
-      val titleHasChanged = potentialNewTitle != entity.title
-      val descriptionHasChanged = potentialNewDescription != entity.description
+      val titleHasChanged = potentialNewTitle != entity.title && potentialNewTitle.nonEmpty
+      val descriptionHasChanged = potentialNewDescription != entity.description && potentialNewDescription.nonEmpty
 
       val updateEntityRequest =
         if (titleHasChanged || descriptionHasChanged) {
           val parentRef = Option.when(folderInfo.expectedParentRef != "")(UUID.fromString(folderInfo.expectedParentRef))
           val updatedTitleOrDescriptionRequest =
             UpdateEntityRequest(
               entity.ref,
-              potentialNewTitle.getOrElse(""),
+              potentialNewTitle.get,
```

Comment:
Is it possible to get an empty `potentialNewTitle` here. If both `folderRow.title` and `entity.title` are None and `descriptionHasChanged` is true, you'll end up trying to get `None.get` Or is it not possible for `entity.title` to be empty? In which case, does it need to be an Option?

Reply:
Because this is an update request, I don't think it should be possible as titles are mandatory when creating the folder; I was thinking if it throws an error then it's a good thing as this is unexpected behaviour. Should I instead do a `folderRow.title.getOrElse(entity.title)` rename it to just "newTitle"?

Reply:
No, lets leave it as is for now. We're trying to second guess what's coming back from Preservica. If we start getting missing titles or security tags then we can rethink it.